src/api.v2/controllers/sampleFileController.js

```diff
@@ -25,19 +25,42 @@ const createFile = async (req, res) => {
     upload_status: 'uploading',
   };
 
-  let uploadUrlParams;
+
 
   await sqlClient.get().transaction(async (trx) => {
     await new SampleFile(trx).create(newSampleFile);
     await new Sample(trx).setNewFile(sampleId, sampleFileId, sampleFileType);
-
-    logger.log(`Getting multipart upload urls for ${experimentId}, sample ${sampleId}, sampleFileType ${sampleFileType}`);
-    uploadUrlParams = await getFileUploadUrls(sampleFileId, metadata, size, bucketNames.SAMPLE_FILES);
   });
 
 
   logger.log(`Finished creating sample file for experiment ${experimentId}, sample ${sampleId}, sampleFileType ${sampleFileType}`);
-  res.json(uploadUrlParams);
+
+  res.json(OK());
+};
+
+const beginUpload = async (req, res) => {
+  const {
+    params: { experimentId, sampleFileId },
+    body: { metadata, size },
+  } = req;
+
+  const { uploadStatus } = await new SampleFile().findById(sampleFileId).first();
```

Comment:
just a random question, isn't it support to just pass the ID in the constructor? something like `new SampleFile(sampleFileId)` to make it less verbose? (no need to change here just to know)

Reply:
Not supported, the constructors can only receive transactions to run with for now.